src/new/wallet.js

```diff
@@ -2469,6 +2464,285 @@ class HathorWallet extends EventEmitter {
 
     return tx;
   }
+
+  /**
+   * Generate addresses and load their history.
+   * The loaded history is added to storage.
+   *
+   * @param {number} startIndex Address index to start to load history
+   * @param {number} count How many addresses I will load
+   *
+   * @return {Promise<void>} Promise that resolves when addresses history is finished loading from server
+   *
+   * @memberof HathorWallet
+   * @inner
+   */
+  async loadAddresses(startIndex, count) {
+    storage.setStore(this.store);
+
+    const addresses = [];
+    const walletData = wallet.getWalletData();
+    const stopIndex = startIndex + count;
+    for (let i = startIndex; i < stopIndex; i++) {
+      // Generate each key from index (if not provided pre-calculated)
+      const address = this.preCalculatedAddresses && this.preCalculatedAddresses[i]
+        ? this.preCalculatedAddresses[i]
+        : wallet.generateAddress(i);
+      walletData.keys[address.toString()] = { index: i };
+      addresses.push(address.toString());
+      // Address derivation can be cpu intensive so we release the event loop for other possible operations
+      await new Promise(resolve => { setTimeout(resolve, 0) });
+    }
+    if (storage.getItem('wallet:address') === null) {
```

Comment:
You can use here `wallet.getCurrentAddress()`, so we prevent this direct storage access.